shared/ui/dropdown/index.tsx

```diff
@@ -0,0 +1,225 @@
+  //   if (!multiple) return [];
+  //   if (!Array.isArray(value)) return [];
+  //   return dropdownItems.filter(item => value.includes(item.value));
+  // };
+
+  return (
+    <View style={styles.wrapper}>
+      {(title || titleTx) && (
+        <Text style={styles.title} tx={titleTx}>
+          {title}
+          {required && <Text style={styles.required}> *</Text>}
+        </Text>
+      )}
+
+      {multiple ? (
+        <ElementMultiSelect
+          style={[
+            styles.dropdown,
+            isFocus && styles.dropdownFocus,
+            error && styles.errorBorder,
+            style,
+          ].filter(Boolean)}
+          placeholderStyle={styles.placeholder}
+          selectedTextStyle={[styles.selectedText, textStyle]}
+          inputSearchStyle={styles.inputSearch}
+          iconStyle={styles.iconStyle}
+          data={dropdownItems}
+          search={search}
+          searchPlaceholder={searchPlaceholder}
+          maxHeight={maxHeight}
+          labelField="label"
+          valueField="value"
+          placeholder={placeholder}
+          value={Array.isArray(value) ? value.map(String) : []}
+          onFocus={() => setIsFocus(true)}
+          onBlur={() => setIsFocus(false)}
+          onChange={selected => {
+            onChange(selected);
+          }}
+          disable={disabled}
+          renderSelectedItem={
+            renderSelectedItems
+              ? item => {
+                  const option = dropdownItems.find(opt => opt.value === item);
+                  if (!option) return null;
+                  // Must return ReactElement or null
+                  const rendered = renderSelectedItems([option]);
+                  // If renderSelectedItems returns an array, return the first element or null
+                  if (Array.isArray(rendered)) return rendered[0] || null;
+                  return rendered || null;
+                }
+              : undefined
+          }
+        />
+      ) : (
+        <ElementDropdown
+          style={[
+            styles.dropdown,
+            isFocus && styles.dropdownFocus,
+            error && styles.errorBorder,
+            style,
+          ].filter(Boolean)}
+          placeholderStyle={styles.placeholder}
+          selectedTextStyle={[styles.selectedText, textStyle]}
+          inputSearchStyle={styles.inputSearch}
+          iconStyle={styles.iconStyle}
+          data={dropdownItems}
+          search={search}
+          searchPlaceholder={searchPlaceholder}
+          maxHeight={maxHeight}
+          labelField="label"
+          valueField="value"
+          placeholder={!isFocus ? placeholder : '...'}
+          value={value}
+          onFocus={() => setIsFocus(true)}
+          onBlur={() => setIsFocus(false)}
+          onChange={item => {
+            if (item) {
+              onChange(item.value);
+              setIsFocus(false);
+            }
+          }}
+          disable={disabled}
+        />
+      )}
```

Comment:
Files exceed 150 LOC. Optimise this code or use external style or split in component